### PR TITLE
⚠️ Add go modules under IPAM api

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
This PR adds go.mod for IPAM api.

**Important:**
Once this PR goes in, those who need to uplift IPAM, and they are only using IPAM APIs , you need to change your go modules to use

```
github.com/metal3-io/ip-address-manager/api
```
 instead of:
```
github.com/metal3-io/ip-address-manager
```

If you are using/referring to other parts of code in IPAM not residing in /api, you need to use both the modules:
```
github.com/metal3-io/ip-address-manager/api
github.com/metal3-io/ip-address-manager
```

### CODE REVIEW

File: Makefile
Comment (furkatgofurov7) [inline Makefile:176]: @maelk I have added the requested changes [asked](https://github.com/metal3-io/ip-address-manager/pull/32#discussion_r630822872) in the previous PR here, PTAL, sorry I had to close the old one. `lint` and `lint-full` also has been added for /api folder. Thanks.

**Review by maelk — APPROVED** [review]

**Comment by maelk** [comment]
Please branch out the version for CAPI v1a3 before merging this to master

**Comment by furkatgofurov7** [comment]
> Please branch out the version for CAPI v1a3 before merging this to master

/hold

We have to do a branching first.

**Comment by furkatgofurov7** [comment]
/test-integration

@kashifest PTAL once again, I had to rebase the PR. Thanks.

**Review by kashifest — APPROVED** [review]

**Comment by furkatgofurov7** [comment]
/hold cancel

We can take this in since branching is in place now.

**Pull request merged**
